### DISCUSSION TITLE
How can I config a sensor and start it with python api?

### DISCUSSION THREAD

**Discussion by wangm23456** [dagster-io/dagster#3555] [Q&A]
How can I config a sensor and start it with python api?
f.e. I want to define a directory_sensor:
```python
@sensor(pipeline_name="log_file_pipeline")
def my_directory_sensor(_context):
    for filename in os.listdir(MY_DIRECTORY):
        filepath = os.path.join(MY_DIRECTORY, filename)
        if os.path.isfile(filepath):
            yield RunRequest(
                run_key=filename,
                run_config={"solids": {"process_file": {"config": {"filename": filename}, 
                                                                  {'other_args': other_args}}}},
            )
```
But how to config the MY_DIRECTORY and other_args dynamically？
If I use a factory to programmatically generate sensor , how start it with python api?

**Comment by gibsondan** [comment]
Hi - using a factory to create sensors seems like a very reasonable option to me.  The @sensor decorator is just a wrapper that creates a SensorDefinition (https://docs.dagster.io/_apidocs/schedules-sensors#dagster.SensorDefinition) - you can absolutely write your own function that creates such a definition dynamically, the resulting SensorDefinition just needs to be included in your repository.

As long as you know the name of your sensor, it doesn't matter if it's created statically in code or dynamically from a factory - you can start it either from dagit or using the `dagster sensor start` CLI command (which I now see is not in the docs, but will add for the release this week - it has identical arguments to `dagster schedule start`). We don't currently have a great way to start it in via calling a Python function, but would the CLI option work?

**Reply by wangm23456** [reply]
So, how can I update the repository? My repository created that using the @repository() decorator in a python file.
